HHS/Head-Start-TTADP
# [TTAHUB-2260] Activity report updates for reduced objective fields

# Stop, potential reviewer! It's recommended by the author that you review #2147 first

## Description of change
- Objective text (for existing objectives) is read-only
- Support type is always editable
- Goal source is editable until goal is on approved AR
- Remove all the code that was displaying previously used Objective topics, etc, as it is dead code

## How to test
- Create an objective on the RTR. Add it to an AR, and confirm that the text is read-only on the AR. Return to the RTR and confirm that it is still read-only, as it has been added to an AR.
- Create an objective on an AR. The text of that objective should be editable on the AR if it was created there
- On the AR, support type is always editable.
- Confirm that goal source is read-only if the goal is on an approved AR. Otherwise, it is editable from a draft AR and from the RTR, and changes in one place are reflected in the other.
- Confirm that for an other entity objective, the objective is editable if it was created on that report, and that support type is always editable.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-2260


## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
